Sources/ResilientDecoding/Resilient.swift

```diff
@@ -62,10 +63,14 @@ public struct Resilient<Value: Decodable>: Decodable {
   func map<T>(transform: (Value) -> T) -> Resilient<T> {
     let value = transform(wrappedValue)
     #if DEBUG
-    return Resilient<T>(
-      value: value,
-      topLevelError: projectedValue.topLevelError,
-      errorsAtOffset: projectedValue.errorsAtOffset)
+    if projectedValue.topLevelError == nil, projectedValue.errorsAtOffset.isEmpty {
```

Comment:
I feel like this kind of code is begging for an enum with associated values to assist with compile-time safety.